realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -223,20 +224,20 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
     public void addingSearchIndexTwice() throws IOException {
         Class[] classes = {PrimaryKeyAsLong.class, PrimaryKeyAsString.class};
 
-        for (final Class clazz : classes){
-            final boolean[] didMigrate = {false};
+        for (final Class clazz : classes) {
+            final AtomicBoolean didMigrate = new AtomicBoolean(false);
 
             RealmMigration migration = new RealmMigration() {
                 @Override
                 public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
-                    Table table = realm.schema.getTable(clazz);
+                    Table table = realm.getSchema().getTable(clazz);
                     long columnIndex = table.getColumnIndex("id");
                     table.addSearchIndex(columnIndex);
-                    if (clazz == PrimaryKeyAsLong.class) {
-                        columnIndex = table.getColumnIndex("name");
-                        table.convertColumnToNullable(columnIndex);
-                    }
-                    didMigrate[0] = true;
+                    // @PrimaryKey fields in PrimaryKeyAsLong and PrimaryKeyAsString.class should be set 'nullable'.
+                    columnIndex = table.getColumnIndex("name");
+                    table.convertColumnToNullable(columnIndex);
+                    //realm.getSchema().getSchemaForClass(clazz).setNullable("name", true);
```

Comment:
Since `Table.convertColumnToNullable(columnIndex)` is an internal API,  `realm.getSchema().getSchemaForClass(clazz).setNullable("name", true)` should be used instead. Somehow, it fails with null pointer exception. I'm working on this to fix and remove it.